### PR TITLE
Hub/git attempt anonymous

### PR DESCRIPTION
**Pull request by Grifs**
## Describe your changes

Attempt github clones in following order:
- HTTPS forced anonymous
- SSH key
- HTTPS with user supplied credentials

## Issue ticket number and link
Closes #xxxx (Replace xxxx with the GitHub issue number)

## Checklist before requesting a review
- [ ] I have performed a self-review of my code

- Does this PR contain:
  - [ ] Breaking changes
  - [ ] New functionality
  - [ ] Major changes
  - [x] Minor changes
  - [ ] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [ ] Relevant unit tests have been added

### CODE REVIEW

**Comment by rcannood** [comment]
Is there something we can do in terms of unit testing?

**Review by rcannood — CHANGES_REQUESTED** [review]
Is there something we can do in terms of unit testing? :)

**Comment by Grifs** [comment]
> Is there something we can do in terms of unit testing? :)

I'm open for suggestions

File: src/main/scala/io/viash/functionality/dependencies/GithubRepository.scala [hunk 1]
```diff
@@ -64,5 +64,21 @@ case class GithubRepository(
...
+      // Checkout with ssh key
+      uri_ssh
+    } else {
+      uri
+    }
+  }
+
   lazy val uri = s"https://github.com/$repo.git"
+  lazy val uri_ssh = s"git@github.com:$repo.git"
+  val fakeCredentials = "nouser:nopass@" // obfuscate the credentials a bit so we don't trigger GitGardian
```
Comment (rcannood) [inline src/main/scala/io/viash/functionality/dependencies/GithubRepository.scala:82]:
```suggestion
  val fakeCredentials = "nouser:nopass@" // obfuscate the credentials a bit so we don't trigger GitGuardian
```

File: src/main/scala/io/viash/functionality/dependencies/ViashhubRepository.scala [hunk 2]
```diff
@@ -64,5 +64,21 @@ case class ViashhubRepository(
...
+      // Checkout with ssh key
+      uri_ssh
+    } else {
+      uri
+    }
+  }
+
   lazy val uri = s"https://viash-hub.com/$repo.git"
+  lazy val uri_ssh = s"git@viash-hub.com:$repo.git"
+  val fakeCredentials = "nouser:nopass@" // obfuscate the credentials a bit so we don't trigger GitGardian
```
Comment (rcannood) [inline src/main/scala/io/viash/functionality/dependencies/ViashhubRepository.scala:82]:
```suggestion
  val fakeCredentials = "nouser:nopass@" // obfuscate the credentials a bit so we don't trigger GitGuardian
```

**Review by rcannood — APPROVED** [review]
Much butter! I could only find a minor typo :)

**Pull request merged**